operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerManager.java

```diff
@@ -41,7 +46,9 @@
 
   @Override
   public void start() throws OperatorException {
-    sources.values().parallelStream().forEach(LifecycleAware::start);
+    // make sure informers are all started before proceeding further
+    ExecutorServiceManager.executeAndWaitForCompletion(
```

Comment:
This can cause a deadlock. For example if the Thread count is 1 (but also others). So the event source is an InformerEventSource and on start it also want to start event sources from the same thread pool. This should not use the same executor service.

Reply:
Not sure I'm following…

Reply:
Lets say there are 2 threads in the in the pool. We start 2 `InformerEventSources` at tha same time. Both started but then both of them starts with the same executor service two InformerManager, but those don't have available threads anymore at that point. Because all the threads available in the executor service are already blocked.

This can be verified if you just create an operator with 1 thread and 1 informer event source, that I assume will never start.